src/cli/README_DATASET.md

```diff
@@ -107,6 +107,108 @@ After the device successfully attaches to a Thread network, the device will retr
    Done
    ```
 
+### Using the Pending Operational Dataset for Delayed Dataset Updates
+
+The Pending Operational Dataset can be used for a delayed update of network parameters on all devices of a Thread Network.
+If certain Active Operational Dataset parameter(s) need to be changed, but the change would impact the connectivity of the network,
+delaying the update helps to let all devices receive the new parameter(s) before the update is applied. Examples of such
+parameters are the channel, PAN ID, certain Security Policy bits, or Network Key.
+
+The delay timer determines the time period after which the Pending Dataset takes effect and becomes the
+Active Dataset. The below example shows how a Pending Dataset with delay timer can be set at a Leader device.
+The Leader will initiate the distribution of the Pending Dataset to the rest of the devices in the network.
+
+Normally, an active Commissioner will set a new Pending Dataset. For testing purposes, we will do this in the example
+directly on the Leader using the CLI - so without using a Commissioner.
```

Comment:
May be useful to add a note here to `dataset updater` CLI command (for `DatasetUpdater` module) as a different way to do the same when on a device.

Edit: (just noticed Jonathan mentioned something similar)

Reply:
+1

Reply:
Agree we could add this note, once the Dataset Updater documentation is included.